xds/src/test/java/io/grpc/xds/sds/trust/SdsX509TrustManagerTest.java

```diff
@@ -221,14 +244,346 @@ public void oneIpAddressInPeerCertsMismatch() throws CertificateException, IOExc
+        return null;
+      }
+
+      @Override
+      public SSLEngineResult unwrap(ByteBuffer byteBuffer, ByteBuffer[] byteBuffers, int i, int i1)
+          throws SSLException {
+        return null;
+      }
+
+      @Override
+      public Runnable getDelegatedTask() {
+        return null;
+      }
+
+      @Override
+      public void closeInbound() throws SSLException {
+
+      }
+
+      @Override
+      public boolean isInboundDone() {
+        return false;
+      }
+
+      @Override
+      public void closeOutbound() {
+
+      }
+
+      @Override
+      public boolean isOutboundDone() {
+        return false;
+      }
+
+      @Override
+      public String[] getSupportedCipherSuites() {
+        return new String[0];
+      }
+
+      @Override
+      public String[] getEnabledCipherSuites() {
+        return new String[0];
+      }
+
+      @Override
+      public void setEnabledCipherSuites(String[] strings) {
+
+      }
+
+      @Override
+      public String[] getSupportedProtocols() {
+        return new String[0];
+      }
+
+      @Override
+      public String[] getEnabledProtocols() {
+        return new String[0];
+      }
+
+      @Override
+      public void setEnabledProtocols(String[] strings) {
+
+      }
+
+      @Override
+      public SSLSession getSession() {
+        return mockSession;
+      }
+
+      @Override
+      public void beginHandshake() throws SSLException {
+
+      }
+
+      @Override
+      public HandshakeStatus getHandshakeStatus() {
+        return null;
+      }
+
+      @Override
+      public void setUseClientMode(boolean b) {
+
+      }
+
+      @Override
+      public boolean getUseClientMode() {
+        return false;
+      }
+
+      @Override
+      public void setNeedClientAuth(boolean b) {
+
+      }
+
+      @Override
+      public boolean getNeedClientAuth() {
+        return false;
+      }
+
+      @Override
+      public void setWantClientAuth(boolean b) {
+
+      }
+
+      @Override
+      public boolean getWantClientAuth() {
+        return false;
+      }
+
+      @Override
+      public void setEnableSessionCreation(boolean b) {
+
+      }
+
+      @Override
+      public boolean getEnableSessionCreation() {
+        return false;
+      }
+
+      @Override
+      public SSLSession getHandshakeSession() {
+        return mockSession;
+      }
+
+      @Override
+      public SSLParameters getSSLParameters() {
+        return sslParameters;
+      }
+
+      @Override
+      public void setSSLParameters(SSLParameters sslParameters) {
+        this.sslParameters = sslParameters;
+      }
+
+      private SSLParameters sslParameters;
+    };
+    sslEngine.setSSLParameters(sslParams);
+    return sslEngine;
+  }
+
+  private SSLSocket buildTrustManagerAndGetSslSocket()
+      throws CertificateException, IOException, CertStoreException {
+    SSLParameters sslParams = buildTrustManagerAndGetSslParameters();
+
+    SSLSocket sslSocket = new SSLSocket() {
```

Comment:
same for this.

Reply:
Mocks won't work here (AFAIK) as stated before.